libs/wire-api/src/Wire/API/Routes/Public/Galley/Feature.hs

```diff
@@ -100,10 +105,14 @@ type FeatureStatusGet f =
     '("get", f)
     (ZUser :> FeatureStatusBaseGet f)
 
-type FeatureStatusPut errs f =
+type family FoldSegments (calls :: [*]) r where
+  FoldSegments '[] r = r
+  FoldSegments (seg ': segs) r = seg :> FoldSegments segs r
```

Comment:
This is the same as the `ApplyMods` type family, right? It would be great to at least leave a comment on that, or even better extract it out in a module and reuse it wherever needed.